test/integration/deployment/deployment_test.go

```diff
@@ -498,3 +499,117 @@ func TestRollbackDeploymentRSNoRevision(t *testing.T) {
+	if err != nil {
+		t.Fatalf("failed to list replicasets of deployments %q and %q in namespace %q: %v", firstDeploymentName, secondDeploymentName, ns.Name, err)
+	}
+	rss := rsList.Items
+	if len(rss) != 2 {
+		t.Fatalf("expected 2 replicasets to be found, but found %d", len(rss))
+	}
+
+	firstController := metav1.GetControllerOf(&rss[0])
+	if firstController == nil {
+		t.Fatalf("replicaset %q is not owned by any deployment", rss[0].Name)
+	}
+	secondController := metav1.GetControllerOf(&rss[1])
+	if secondController == nil {
+		t.Fatalf("replicaset %q is not owned by any deployment", rss[1].Name)
+	}
+
+	// Set up [controllerRef's UID: replicaset] mapping for future usage
+	mapping := map[types.UID]*v1beta1.ReplicaSet{firstController.UID: &rss[0], secondController.UID: &rss[1]}
+
+	// Verify replicasets are owned by the created deployments
+	for _, tester := range testers {
+		if _, ok := mapping[tester.deployment.UID]; !ok {
+			t.Fatalf("expected deployment %q owns replicaset %q or %q, but it does not", tester.deployment.Name, rss[0].Name, rss[1].Name)
+		}
+	}
+
+	// Scale only the first deployment by 1
+	newReplicas := replicas + 1
+	testers[0].deployment, err = testers[0].updateDeployment(func(update *v1beta1.Deployment) {
+		update.Spec.Replicas = &newReplicas
+	})
+	if err != nil {
+		t.Fatalf("failed updating deployment %q: %v", firstDeploymentName, err)
+	}
+
+	// Wait for the controller to notice the scale
+	if err := testers[0].waitForObservedDeployment(testers[0].deployment.Generation); err != nil {
+		t.Fatalf("encountered an error when controller waits to notice the scale: %v", err)
+	}
+
+	// Verify that replicaset of first deployment is scaled
+	rs, _ := mapping[testers[0].deployment.UID]
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get replicaset %q: %v", rs.Name, err)
+	}
+	if *rs.Spec.Replicas != newReplicas {
+		t.Errorf("expected replicaset %q of deployment %q is scaled from %d to %d, but found %d replicas", rs.Name, firstDeploymentName, replicas, newReplicas, *rs.Spec.Replicas)
+	}
+
+	// Verify that replicaset of second deployment is not scaled
+	rs, _ = mapping[testers[1].deployment.UID]
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get replicaset %q: %v", rs.Name, err)
+	}
+	if *rs.Spec.Replicas != replicas {
+		t.Errorf("expected replicaset %q of deployment %q still has %d replicas, but found %d replicas", rs.Name, secondDeploymentName, replicas, *rs.Spec.Replicas)
+	}
+
+	// Make sure the deployment completes while manually marking deployment pods as ready at the same time
+	// Use soft check because this deployment was just scaled and rolling update strategy might be violated
+	if err := testers[0].waitForDeploymentCompleteAndMarkPodsReady(); err != nil {
```

Comment:
optional: this can move to after `waitForObservedDeployment()` and the above replicas check or just replace `waitForObservedDeployment()`